addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3036,6 +3036,37 @@ registry.many2one = SnippetOptionWidget.extend({
     }
 });
 
+/**
+ * Allows to display a warning message on outdatedSnippets
+ */
+registry.VersionControl = SnippetOptionWidget.extend({
+    xmlDependencies: ['/web_editor/static/src/xml/snippets.xml'],
+    /**
+     * @override
+     */
+    start: function () {
+        if (!this._isUpToDate()) {
+            $(qweb.render('web_editor.outdated_block_message')).prependTo(this.$el);
+        }
+        this._super(...arguments);
+    },
+    /**
+     * Returns whether the snippet is up to date with the current version or not
+     *
+     * @private
+     * @returns {boolean}
+     */
+    _isUpToDate: function () {
+        const targetName = this.$target[0].dataset.snippet;
+        const referenceSection = document.querySelector(`#o_scroll .oe_snippet > [data-snippet="${targetName}"]`);
```

Comment:
Don't search the document, let's do things properly: trigger_up an event with the snippet name as param, then let the snippet menu catch it and return you an object with vjs, vscss and vxml.

Reply:
When you say return you mean using a onSuccess method ?

Reply:
Of course